core/point/point.go

```diff
@@ -17,7 +44,48 @@ func New(x, y int64) *Point {
 }
 
 // X returns the x-value.
-func (p *Point) X() int64 { return p.x }
+func (pt *Point) X() int64 { return pt.x }
 
 // Y returns the y-value.
-func (p *Point) Y() int64 { return p.y }
+func (pt *Point) Y() int64 { return pt.y }
+
+// ToSGF converts a pointer-type (immutable) *Point to an SGF Point (
+// two letter string).
+// The returned value is 0-indexed.
+func (pt *Point) ToSGF() (string, error) {
+	var sgfOut string
+	var err01 = fmt.Errorf("")
+	if (0 <= pt.X()) && (pt.X() <= 51) && (0 <= pt.Y()) && (pt.Y() <= 51) {
+		sgfX := string(pointToSgfMap[pt.X()])
+		sgfY := string(pointToSgfMap[pt.Y()])
+		sgfOut = sgfX + sgfY
+		err01 = nil
+	} else {
+		sgfOut = ""
+		err01 = fmt.Errorf("*Point int64 x and y value entries must" +
+			" be greater than or equal to 0, " +
+			"and less than or equal to 51. ")
+	}
+	return sgfOut, err01
+}
+
+// NewFromSGF converts an SGF point (
+// two letter string) to a pointer-type (immutable) *Point (
+// (struct with two rune/char values).
+func NewFromSGF(sgfPt string) (*Point, error) {
+	var intX int64
+	var intY int64
+	var err02 = fmt.Errorf("")
+	if (sgfPt != "") && (len(sgfPt) == 2) {
```

Comment:
For this code block, you can do just

```
return New(intX, intY, nil)
```

However, the logic should be reversed.

```golang
if ...check error cases... {
  return nil, fmt.Errorf("some error")
}
do non-error stuff
```